rewrite-core/src/main/java/org/openrewrite/config/Environment.java

```diff
@@ -37,9 +37,17 @@ public Collection<Recipe> listRecipes() {
         List<Recipe> recipes = resourceLoaders.stream()
                 .flatMap(r -> r.listRecipes().stream())
                 .collect(toList());
+        Set<Recipe> recipeExclusions = new HashSet<>();
         for (Recipe recipe : recipes) {
             if (recipe instanceof DeclarativeRecipe) {
                 ((DeclarativeRecipe) recipe).initialize(recipes);
+                recipeExclusions.addAll(((DeclarativeRecipe) recipe).getExcludeRecipeList());
+            }
+        }
+
+        if (!recipeExclusions.isEmpty()) {
+            for (Recipe recipe : recipes) {
+                recipe.getRecipeList().removeAll(recipeExclusions);
```

Comment:
For attention: recipes that have options need an `@EqualsAndHashCode(callSuper = true)`